### DISCUSSION TITLE
DOCS: RLS with custom JWT

### DISCUSSION THREAD

**Discussion by tg44** [supabase/supabase#12271] [General]
# RLS with custom JWT

## Link 

[RLS page](https://supabase.com/docs/guides/auth/row-level-security)

## Describe the problem

The custom JWT part is missing.

## Describe the improvement

I would add a section which at least mentions that you can do some RLS with reading the JWT claim;
```sql
create policy "Messages viewable by server members."
    on messages for select
          using (
             ((server_id)::text = ((current_setting('request.jwt.claims'::text, true))::json ->> 'serverId'::text))
         );
```

# Other relevant links 
https://github.com/supabase/supabase/issues/6290
https://stackoverflow.com/questions/67551593/supabase-client-permission-denied-for-schema-public
https://github.com/supabase/supabase/discussions/1148
https://medium.com/@gracew/using-supabase-rls-with-a-custom-auth-provider-b31564172d5d
(I also found multiple reddit questions about these topics.)

**Comment by kiwicopple** [comment]
Hey @tg44 - there is a lot going on here! Am i right that this is 3 different issues? Would you mind splitting them up so we can tackle them each separately?

**Comment by tg44** [comment]
@kiwicopple done!
